conda_build/noarch_python.py

```diff
@@ -24,7 +26,7 @@ def rewrite_script(fn):
     directory after it passes some sanity checks for noarch pacakges"""
 
     # Load and check the source file for not being a binary
-    src = join(config.build_prefix, 'bin', fn)
+    src = join(config.build_prefix, 'Scripts' if ISWIN else 'bin', fn)
```

Comment:
Or maybe we should just check for .exe as you suggest below. And raise an error there, unless its an exe `foo.exe` that has a matching `foo-script.py`.